docs/QbeastFormat.md

```diff
@@ -182,9 +224,63 @@ ConvertToQbeastCommand(tableIdentifier, columnsToIndex, desiredCubeSize).run(spa
 val qTable = spark.read.format("qbeast").load(path)
 ```
 By doing so, we also enable subsequent appends using either delta or qbeast.
-Conversion on a partitioned table is not supported.
 
-### Compaction
+> Conversion on a partitioned table is not supported.
+
+
+## Optimization
+
+**Optimize** is an expensive operation that consist on:
+- moving offsets to their legitimate cubes (files).
+- rolling-up small files into their parents, creating new files with a size closer to the `desiredCubeSize`.
+
+Thus accomplishing a **better file layout and query performance**.
+
+To minimize write amplification of this command, **we execute it based on subsets of the table**, like `Revision ID's` or specific files.
+
+`Revisions` determine the area of the index. **The last Revision would contain the whole dataset area**, while previous Revisions would contain a _subset_ of the area written before.
+It can be useful to optimize by specific Revision ID's when we have an incremental appends scenario.
+
+As an example, imagine you have a table called `city_taxi` that contains data about location of taxi stations of one or more countries.
+
+An initial commit C1 is inserting data into the table about taxis in France, assuming a column called `city_name` indexed with Qbeast.
+With this first commit C1, we add three files F1, F2 and F3.
+
+> The first commit C1 creates a revision R1. The space contained on revision R1 is `[city_name_min = France, city_name_max = France]`
+
+A second commit `C2` adds data about German cities, so it expands the cardinality of the `city_name` column.
+This commit adds another four files F4, F5, F6 and F7.
+
+>Now, C2 creates revision R2. The space of the revision R2 is `[city_name_min = France, city_name_max = Germany]`
+
+Revision `R2` is necessary because it expands the cardinality of the city name dimension, and this expansion impacts the index as the range of values changes.
+
+Optimizing consequently **can be performed over either revision**.
+> Running optimize over R1 would rewrite (only rearranging data) files F1, F2 and F3.
+
+### Optimize API
+These are the 3 ways of executing the `optimize` operation:
+
+```scala
+qbeastTable.optimize() // Optimizes the last Revision Available.
+// This does NOT include previous Revision's optimizations.
+
+qbeastTable.optimize(2L) // Optimizes the Revision number 2.
+
+qbeastTable.optimize(Seq("file1", "file2")) // Optimizes the specific files
+```
+
+**If you want to optimize the full table, you must loop through `revisions`**:
+
+```scala
+val revisions = qbeastTable.revisionsIDs() // Get all the Revision ID's available in the table.
+revisions.foreach(revision => 
+  qbeastTable.optimize(revision)
+)
+```
+> Note that **Revision ID number 0 is reserved for Stagin Area** (non-indexed files). This ensures compatibility with underlying table formats.
+
+## Compaction
```

Comment:
I understand the general compaction concept, but can we elaborate on what it is doing for us specifically here?

Reply:
Sure! I will also add that the behaviour is different than previous 0.6.0 version (now calls optimize() underneath) and will reference issue #294